WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -308,12 +314,24 @@ private async Task SignTransactionAsync(IEnumerable<AliceClient> aliceClients, T
 		}
 	}
 
-	private async Task ReadyToSignAsync(IEnumerable<AliceClient> aliceClients, CancellationToken cancellationToken)
+	private async Task ReadyToSignAsync(IEnumerable<AliceClient> aliceClients, DateTimeOffset readyToSignEndTime, CancellationToken cancellationToken)
 	{
-		foreach (var aliceClient in aliceClients)
+		var remainingTime = readyToSignEndTime - DateTimeOffset.UtcNow;
+		if (remainingTime < TimeSpan.FromSeconds(3))
 		{
-			await aliceClient.ReadyToSignAsync(cancellationToken).ConfigureAwait(false);
+			throw new InvalidOperationException("No time to signal ready, aborting.");
 		}
```

Comment:
Why?

Reply:
Sanity check and having more logs about failures. If this exception comes we will know that the client had no practical chance to finish. Otherwise, we will just get an unexpected round phase message. However, the backend will indicate the error with a very clear message IMO - so taking this into account does not make a lot of sense. I will remove it if u agree.